yarn-project/end-to-end/src/integration_l1_publisher.test.ts

```diff
@@ -9,45 +11,92 @@ import {
   range,
 } from '@aztec/circuits.js';
 import { fr, makeNewContractData, makeProof } from '@aztec/circuits.js/factories';
-import { EthereumRpc } from '@aztec/ethereum.js/eth_rpc';
-import { WalletProvider } from '@aztec/ethereum.js/provider';
-import { DecoderHelper, Rollup, UnverifiedDataEmitter } from '@aztec/l1-contracts';
+import { createDebugLogger } from '@aztec/foundation/log';
+import { DecoderHelperAbi, RollupAbi } from '@aztec/l1-artifacts';
+import {
+  EmptyRollupProver,
+  L1Publisher,
+  SoloBlockBuilder,
+  WasmRollupCircuitSimulator,
+  getCombinedHistoricTreeRoots,
+  getL1Publisher,
+  getVerificationKeys,
+  makeEmptyProcessedTx as makeEmptyProcessedTxFromHistoricTreeRoots,
+  makeProcessedTx,
+  makePublicTx,
+} from '@aztec/sequencer-client';
 import { MerkleTreeOperations, MerkleTrees } from '@aztec/world-state';
 import { beforeAll, describe, expect, it } from '@jest/globals';
 import { default as levelup } from 'levelup';
-import { SoloBlockBuilder } from '../block_builder/solo_block_builder.js';
-import { createMemDown } from '../block_builder/solo_block_builder.test.js';
-import { getVerificationKeys, makePublicTx } from '../index.js';
-import { EmptyRollupProver } from '../prover/empty.js';
-import { EthereumjsTxSender } from '../publisher/ethereumjs-tx-sender.js';
-import { L1Publisher } from '../publisher/l1-publisher.js';
 import {
-  makeEmptyProcessedTx as makeEmptyProcessedTxFromHistoricTreeRoots,
-  makeProcessedTx,
-} from '../sequencer/processed_tx.js';
-import { getCombinedHistoricTreeRoots } from '../sequencer/utils.js';
-import { WasmRollupCircuitSimulator } from '../simulator/rollup.js';
-import { hexStringToBuffer } from '../utils.js';
+  Address,
+  Chain,
+  GetContractReturnType,
+  HttpTransport,
+  PublicClient,
+  createPublicClient,
+  encodeFunctionData,
+  getAbiItem,
+  getAddress,
+  getContract,
+  http,
+} from 'viem';
+import { privateKeyToAccount } from 'viem/accounts';
+import { foundry } from 'viem/chains';
+import { deployL1Contracts } from './deploy_l1_contracts.js';
 
 // Accounts 4 and 5 of Anvil default startup with mnemonic: 'test test test test test test test test test test test junk'
 const sequencerPK = '0x47e179ec197488593b187f80a00eb0da91f1b9d0b13f8733639f19c30a34926a';
 const deployerPK = '0x8b3a350cf5c34c9194ca85829a2df0ec3153be0318b5e2d3348e872092edffba';
-const anvilHost = process.env.ANVIL_HOST ?? 'http://127.0.0.1:8545';
-const chainId = 31337;
-
-describe.skip('L1Publisher integration', () => {
-  let decoderHelper: DecoderHelper;
-  let rollup: Rollup;
-  let unverifiedDataEmitter: UnverifiedDataEmitter;
-  let ethRpc: EthereumRpc;
+
+const logger = createDebugLogger('aztec:integration_l1_publisher');
+
+const config = getConfigEnvVars();
+
+describe('L1Publisher integration', () => {
+  let publicClient: PublicClient<HttpTransport, Chain>;
+
+  let rollupAddress: Address;
```

Comment:
Both having `Address` and `EthAddress` can be a little confusing, but guess that is just the way it is.

Reply:
Yes, it's a mess. I think that if it turns out that we really like viem I would simply replace EthAddress with the viem's type

Reply:
Would be nice with the consistency 👍